### DISCUSSION TITLE
What's up with all the different cursors?

### DISCUSSION THREAD

**Discussion by mycroftcanner** [sbertix/Swiftagram#236] [Q&A] [answered]
HI all,

Just trying to understand the difference between the different cursors returned when getting the Inbox.

The raw response contains:
`oldestCursor`
`newestCursor`
`previousCursor`
`nextCursor`

In `Swiftagram` we are using the `oldestCursor` to paginate. Is there any particular reason why? What is the difference between all of these? Intuitively, I would have expected `nextCursor` to be the one required.

Thanks

**Comment by sbertix** [comment, marked as answer]
Hey @mycroftcanner, 

I'm assuming you're referring to direct related `Endpoint`s. 
The simplest response would be: "cause Instagram does this themselves". 
Without going into much detail, let's take threads pagination as an example, as the mechanism for the inbox is slightly more complicated as there's a lot of boilerplate code in the response that is never actually used (I'm assuming it comes down to the several iterations of the messaging API): using "oldest" and "newest" simply guarantees time-based sorting. Most times they'd be the same (as messages cannot be scheduled) but it's a neat bonus.
